pkg/server/autoconfig/task_markers.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package autoconfig
+
+import (
+	"bytes"
+	"context"
+	"strconv"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/util/encoding"
+	"github.com/cockroachdb/errors"
+)
+
+// InfoKeyCompletionPrefix is the prefix of the key inserted in job_info
+// when a task has completed.
+var InfoKeyCompletionPrefix = roachpb.Key("completed-")
+
+// InfoKeyStartPrefix is the prefix of the key inserted in job_info
+// when a task has started.
+var InfoKeyStartPrefix = roachpb.Key("started-")
+
+// DecodeTaskID returns the task ID for the given infoKey.
+func DecodeTaskID(prefix roachpb.Key, infoKey []byte) (uint64, error) {
+	if !bytes.HasPrefix(infoKey, []byte(prefix)) {
+		return 0, errors.AssertionFailedf("programming error: prefix %q missing: %q", string(prefix), string(infoKey))
+	}
+	infoKey = infoKey[len(prefix):]
+	_, v, err := encoding.DecodeUvarintAscending(infoKey)
+	return v, err
+}
+
+// MakeTaskInfoKey creates a job_info infoKey for the given task.
+func MakeTaskInfoKey(prefix roachpb.Key, taskID uint64) roachpb.Key {
+	infoKey := make([]byte, 0, len(prefix)+10)
+	infoKey = append(infoKey, []byte(prefix)...)
+	infoKey = encoding.EncodeUvarintAscending(infoKey, taskID)
+	return infoKey
+}
+
+// writeStartMarker writes a start marker for the given task ID and
+// also writes its job ID into the value part.
+func writeStartMarker(
```

Comment:
why are you passing an `execCfg` and then not using it?

I view an execCfg argument to a function as a code smell